src/techs/dto/update-techstack-category.dto.ts

```diff
@@ -0,0 +1,24 @@
+import { ApiProperty } from "@nestjs/swagger";
+import { IsInt, IsNotEmpty, IsString } from "class-validator";
+
+export class UpdateTechStackCategoryDto {
+    @IsString()
+    @IsNotEmpty()
+    @ApiProperty({ example: "ORM" })
+    newName: string;
+
+    @IsString()
+    @IsNotEmpty()
+    @ApiProperty({ example: "DB interface" })
+    description: string;
+
+    @IsInt()
+    @IsNotEmpty()
+    @ApiProperty({ example: 1 })
+    categoryId: number;
+
+    @IsInt()
+    @IsNotEmpty()
+    @ApiProperty({ example: 1 })
+    voyageTeamId: number;
```

Comment:
voyageTeamId is used here to verify the user's "own team" and then to verify that the category belongs to that team
![image](https://github.com/user-attachments/assets/0ec7ba8e-5146-4d5c-a6a7-ba95ba5c1c08)

Reply:
Ok - I have the other revisions completed.  After going over this suggestion, I think voyageTeamId needs to be stay in the DTO or be included in the URI.  This route checks for duplicate categories before posting to db, and it needs the category name and voyageTeamId to identify them.

Reply:
I think we can have the categoryId in the uri as it's identifying the resource. 

I think we won't need the teamId to check - check out `src/ability/conditions/resource.ability.ts`, would something like this work?

Reply:
The teamId isn't used for permissions, it's needed to identify duplicate categories.  Because category names aren't unique anymore, they need to be matched by name and teamId

Reply:
I see. I was thinking we can find the teamId using the categoryId. Then use that, unless they gave the wrong teamId, which I think they would get a 403 anyway...

![image](https://github.com/user-attachments/assets/08b5a4ac-3ae2-4300-998c-cba334d299f5)

but then it looks like it would add another db query, I don't think it's a big issue to require an extra input, but then every extra input we'll have to do validation, which I think we already have anyway. 

on another note: 

we could also use `@@unique` teamId + CategoryId, then insert without checking since it would just throw an error if the combination is not unique, which would save one db query based on your current implementation

Reply:
I also think you'll need to make teamId not nullable, that's why I also asked that question if it needs to be optional. Now I remember it was because we had existing data and we decided to make it nullable to accomodate old data

Reply:
One problem with using `@@unique[ name, teamId]` is that it won't use case-insensitive mode for the string comparison, so we might have to drop that

Reply:
> Do you think citext would work?

Yeah that looks good, and plus it's already installed
Would I use it the same way as in this example?
![image](https://github.com/user-attachments/assets/456771ed-41d2-42e8-bccb-2b4298050314)

Reply:
yes I think so